crates/driver/src/domain/competition/auction.rs

```diff
@@ -48,38 +71,23 @@ impl From<chrono::DateTime<chrono::Utc>> for Deadline {
 /// most optimal solution, but still ensure there is time left for the
 /// driver to forward the results back to the protocol or do some other
 /// necessary work.
-///
-/// This type contains a [`std::time::Duration`] rather than
-/// [`chrono::Duration`] because [`std::time::Duration`] is guaranteed
-/// to be nonnegative, while [`chrono::Duration`] can be negative as well.
 #[derive(Debug, Clone, Copy)]
-pub struct SolverDeadline(std::time::Duration);
-
-impl From<SolverDeadline> for std::time::Duration {
-    fn from(deadline: SolverDeadline) -> Self {
-        deadline.0
-    }
-}
+pub struct SolverDeadline(chrono::DateTime<chrono::Utc>);
 
 impl From<SolverDeadline> for chrono::DateTime<chrono::Utc> {
-    fn from(deadline: SolverDeadline) -> Self {
-        chrono::Utc::now()
-            .checked_add_signed(chrono::Duration::from_std(deadline.0).unwrap())
-            .unwrap()
+    fn from(value: SolverDeadline) -> Self {
+        value.0
     }
 }
 
-impl Deadline {
-    pub fn for_solver(&self) -> Result<SolverDeadline, DeadlineExceeded> {
-        let timeout = self.0 - chrono::Utc::now() - Self::time_buffer();
-        timeout
-            .to_std()
-            .map(SolverDeadline)
-            .map_err(|_| DeadlineExceeded)
-    }
-
-    fn time_buffer() -> chrono::Duration {
-        chrono::Duration::seconds(1)
+impl SolverDeadline {
+    pub fn timeout(&self, now: time::Now) -> Result<std::time::Duration, DeadlineExceeded> {
+        let timeout = self.0 - now.now();
+        if timeout <= chrono::Duration::zero() {
+            Err(DeadlineExceeded)
+        } else {
+            Ok(timeout.to_std().expect("already checked non-negative"))
+        }
```

Comment:
You can make this more succinct with:
```suggestion
        timeout.to_std().map_err(|_| DeadlineExceeded)
```